app/react/portainer/registries/ListView/DefaultRegistry/DefaultRegistryAction.tsx

```diff
@@ -0,0 +1,77 @@
+import {
+  usePublicSettings,
+  useUpdateDefaultRegistrySettingsMutation,
+} from 'Portainer/settings/queries';
+import { notifySuccess } from 'Portainer/services/notifications';
+import { FeatureId } from 'Portainer/feature-flags/enums';
+
+import { Tooltip } from '@@/Tip/Tooltip';
+import { Button } from '@@/buttons';
+import { Icon } from '@@/Icon';
+import { BEFeatureIndicator } from '@@/BEFeatureIndicator';
+
+export function DefaultRegistryAction() {
+  const settingsQuery = usePublicSettings({
+    select: (settings) => settings.DefaultRegistry?.Hide,
+  });
+  const defaultRegistryMutation = useUpdateDefaultRegistrySettingsMutation();
+
+  if (!settingsQuery.isSuccess) {
+    return null;
+  }
+  const hideDefaultRegistry = settingsQuery.data;
+
+  return (
+    <>
+      {!hideDefaultRegistry ? (
+        <div className="vertical-center">
+          <Button
+            className="btn btn-xs btn-light vertical-center"
+            onClick={() => handleShowOrHide(true)}
+            disabled
```

Comment:
this is only disabled if the feature is disabled. if you move this code to EE (as required), it will be disabled too

need to do something like this:

```
import {isLimitedToBe} from '@/app/portainer/feature-flags/feature-flags.service'

...
function DefaultRegistryAction() {
	const isLimited = isLimitedToBe(FeatureId.HIDE_DOCKER_HUB_ANONYMOUS)

	...

	<Button

		disabled={isLimited}

....


```

Reply:
fixed